### PR TITLE
Argument tests for CreateJsonTypeInfo and CreateJsonPropertyInfo

### LINKED ISSUE #71296: CreateJsonPropertyInfo NullReferenceException
### Description

CreateJsonPropertyInfo throws NRE when type is null.

cc @eiriktsarpalis 

### Reproduction Steps

```cs
JsonTypeInfo.CreateJsonPropertyInfo(null, "Name");
```

### Expected behavior

ArgumentNullException

### Actual behavior

NullReferenceException

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/runtime/issues/71296

I went through all APIs and seems only CreateJsonTypeInfo and CreatePropertyInfo were missing checks. I also added some test gaps for null modifier but that was checking args already

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs [hunk 1]
```diff
@@ -427,6 +432,21 @@ public static JsonTypeInfo<T> CreateJsonTypeInfo<T>(JsonSerializerOptions option
...
         {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs:443]: We should be using `ArgumentNullException.ThrowIfnull` for cases like that, see #68178.

**Review by eiriktsarpalis — APPROVED** [review]
Other than one minor nit, lgtm.

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs [hunk 2]
```diff
@@ -440,7 +460,20 @@ public static JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions o
...
+
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (IsInvalidForSerialization(propertyType))
+            {
+                ThrowHelper.ThrowArgumentException_CannotSerializeInvalidType(nameof(propertyType), propertyType, Type, name);
+            }
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs:476]: Does the method check if `TypeInfoKind == JsonTypeInfoKind.Object`. We migth want to throw `InvalidOperationException` if we call the create method for an invalid declaring type.

Comment (krwq) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs:476]: I thought we discussed this and decided not to throw. We won't let you add such property but no problem with creating

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs [hunk 3]
```diff
@@ -580,6 +580,54 @@ public static void AddingNullJsonPropertyInfoIsNotPossible()
...
+        [Theory]
+        [InlineData(typeof(void))]
+        [InlineData(typeof(Dictionary<,>))]
+        [InlineData(typeof(List<>))]
+        [InlineData(typeof(Nullable<>))]
+        [InlineData(typeof(int*))]
+        [InlineData(typeof(RefStruct))]
+        public static void CreateJsonTypeInfoWithInappropriateTypeThrows(Type type)
+        {
+            Assert.Throws<ArgumentException>(() => JsonTypeInfo.CreateJsonTypeInfo(type, new JsonSerializerOptions()));
```
Comment (steveisok) [inline src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs:601]: @fanyang-mono  One of the types being passed in (probably void) results in a runtime crash.  Would you be able to take a look and see why?

Comment (krwq) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs:601]: I've filed https://github.com/dotnet/runtime/issues/71339 as I'm most likely going to try to workaround the issue in this PR

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs (hunk 1 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs:443]: Yeah, I just realized. We have an equivalent method in `ThrowHelper` so you can just use that.

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs (hunk 2 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs:476]: Fair enough -- probably not too important.

File: src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs (hunk 3 above)
Comment (fanyang-mono) [reply src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonTypeInfo.cs:601]: I missed the message yesterday. Will take a look at the issue when I get a chance.

**Pull request merged**